### PR TITLE
Fix: Not loosing the access token when calling UserCredentials#ToBuil…

### LINKED ISSUE #992: Access token lost in transit, when using UserCredentials ToBuilder()
#### Environment details

When using the library, I found a bug in the UserCredentials#ToBuilder()
It's not calling super and therefore we lost the access token

#### Steps to reproduce

```
UserCredentialsTest.java

  @Test
  public void toBuilder() throws IOException {
    final URI tokenServer1 = URI.create("https://foo1.com/bar");
    AccessToken accessToken = new AccessToken(ACCESS_TOKEN, null);
    MockHttpTransportFactory httpTransportFactory = new MockHttpTransportFactory();
    UserCredentials credentials =
            UserCredentials.newBuilder()
                    .setClientId(CLIENT_ID)
                    .setClientSecret(CLIENT_SECRET)
                    .setRefreshToken(REFRESH_TOKEN)
                    .setAccessToken(accessToken)
                    .setHttpTransportFactory(httpTransportFactory)
                    .setTokenServerUri(tokenServer1)
                    .build();

    credentials.toBuilder().build();
    UserCredentials otherCredentials = credentials.toBuilder().build();
    assertEquals(credentials, otherCredentials);
  }
```

Will shortly provide a PR for this

### PR DESCRIPTION
**Pull request by qcastel**
Fixes #992 ☕️

### CODE REVIEW

File: oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java [hunk 1]
```diff
@@ -830,6 +830,27 @@ public void IdTokenCredentials_NoUserEmailScope_throws() throws IOException {
     }
   }
 
+  @Test
+  public void toBuilder() throws IOException {
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java:834]: Please rename similarly to other tests, so the name describes what is being tested and expected behavior. 
Like UserCredentians_ToBuilder_AccessTokenSaved

File: oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java [hunk 2]
```diff
@@ -830,6 +830,27 @@ public void IdTokenCredentials_NoUserEmailScope_throws() throws IOException {
...
+            UserCredentials.newBuilder()
+                    .setClientId(CLIENT_ID)
+                    .setClientSecret(CLIENT_SECRET)
+                    .setRefreshToken(REFRESH_TOKEN)
+                    .setAccessToken(accessToken)
+                    .setHttpTransportFactory(httpTransportFactory)
+                    .setTokenServerUri(tokenServer1)
+                    .build();
+
+    credentials.toBuilder().build();
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java:848]: This line is not needed?

**Review by TimurSadykov — CHANGES_REQUESTED** [review]
Thanks for the contribution! Few minor comments.

**Comment by qcastel** [comment]
@TimurSadykov As I was making the test a little bit more generic than just checking the access token, I found out that two methods of the builders were missing.
The test is now covering that every attributes of the credentials are copied over correctly during the ToBuilder()

**Comment by qcastel** [comment]
@TimurSadykov any news on this one?

File: oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java [hunk 3]
```diff
@@ -830,6 +833,25 @@ public void IdTokenCredentials_NoUserEmailScope_throws() throws IOException {
     }
   }
 
+  @Test
+  public void UserCredentials_ToBuilder_CopyEveryAttributes() {
```
Comment (TimurSadykov) [inline oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java:837]: nit, every part has to stat with lowercase, like this and a typo fix: userCredentials_toBuilder_copyEveryAttribute

**Review by TimurSadykov — APPROVED** [review]
Sorry for a delay. 

LGTM, i will merge once the last nit is fixed.

File: oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java (hunk 3 above)
Comment (qcastel) [inline oauth2_http/javatests/com/google/auth/oauth2/UserCredentialsTest.java:837]:
```suggestion
  public void userCredentials_toBuilder_copyEveryAttribute() {
```

**Review by TimurSadykov — APPROVED** [review]
LGTM

**Pull request merged**